frontend_tests/node_tests/typeahead.js

```diff
@@ -52,6 +52,8 @@ run_test("get_emoji_matcher", () => {
 
     assert_matches("japanese_post_", [emoji_japanese_post_office]);
     assert_matches("japanese post ", [emoji_japanese_post_office]);
+
+    assert_matches("🐼", [emoji_panda_face]);
```

Comment:
Just to be clear, is this example that the user would type `:🐼` in the compose box?

Reply:
Correct. I had thought that it would also work in the emoji picker used for reactions and setting emoji status, but it appears that a different matching algorithm is used: 
https://github.com/Fingel/zulip/blob/main/static/js/emoji_picker.js#L238
Perhaps it wouldn't be too hard to reconcile them both?

The use case for matching a emoji directly is more common in mobile, since many keyboards have emoji support built in, some people even use full emoji keyboards. But since it does seem like correct behavior regardless even for web, I thought it couldn't hurt to move it into shared.

Reply:
In the compose box, I think people will naturally type simply `🐼`. I'm not sure there's an expectation to be able to type `:🐼` and have that mean anything.

Where I think this functionality really makes a difference is in choosing an emoji reaction. Here's the original issue for it in mobile: https://github.com/zulip/zulip-mobile/issues/3961 I think it's pretty natural to expect that if you're prompted for an emoji, and you enter an actual emoji, that the software ought to recognize that as meaning that emoji.

Probably the main use case for this in the web app is in mobile web, because phones tend to have nice emoji keyboards.

Reply:
OK, so it sounds like we might need a follow-up issue for the web app's reaction emoji picker to support pasting in an emoji character (for use on mobile web)?  I think that's a fine state to leave things in, as it's not a regression.

Reply:
Sounds right to me. A secondary benefit of that follow-up PR might be to use consistent logic between the colon emoji typeahead and the emoji-picker typeahead, if that is desired.

Reply:
I think those might be intentionally different; but I don't know for sure the details.

Reply:
They appear to both use `typeahead.sort_emojis`, but the matching is a little different; the emoji picker does this:

```
            for (const emoji_dict of emojis) {                                                           
                for (const alias of emoji_dict.aliases) {                                                
                    const match = search_terms.every((search_term) => alias.includes(search_term));      
                    if (match) {                                                                         
                        search_results.push({...emoji_dict, emoji_name: alias});                         
                        break; // We only need the first matching alias per emoji.                       
                    }                                                                                    
                }                                                                                        
            }                                                                                            
```

Possible differences include (1) how aliases are handled, (2) the special `_/ ` handling in the compose box and (3) the emoji picker handling inputting multiple words without any consideration for their order in the filter box due to its different context.

I'm not sure which of those are true differences.